virttest/utils_misc.py

```diff
@@ -2564,12 +2564,18 @@ def __get_unit_index(M):
         raise ValueError("Invalid data size format 'value_str=%s'" % value_str)
     from_index = __get_unit_index(unit)
     to_index = __get_unit_index(order_magnitude)
-    scale = int(factor) ** (to_index - from_index)
-    if scale > 0:
-        data_size = float(value) / abs(scale)
+    scale = int(factor) ** abs(to_index - from_index)
+    if to_index == from_index:
+        data_size = float(value)
```

Comment:
@xutian 
>if factor 1000 not is default 1024 , results is unexpected.

This should be the same result as ```(to_index - from_index) = 0```
>And I'm not see anything wrong when convert from small unit to big unit.

Yes, you are right. The problem here is ```scale = int(factor) ** (to_index - from_index)``` will always greater than 0, so it will always go to the following ```if``` block.